release notes/v0.40.0.md

```diff
@@ -0,0 +1,253 @@
+      }
+      else {
+        console.log(`VU ${__VU}:${id} received unhandled message: ${msg.message}`)
+      }
+    })
+
+
+    let intervalId = setInterval(() => {
+      ws.send(JSON.stringify({ 'event': 'SAY', 'message': `I'm saying ${randomString(5)}` }));
+    }, randomIntBetween(2000, 8000)); // say something every 2-8seconds
+
+
+    let timeout1id = setTimeout(function() {
+      clearInterval(intervalId)
+      console.log(`VU ${__VU}:${id}: ${sessionDuration}ms passed, leaving the chat`);
+      ws.send(JSON.stringify({ 'event': 'LEAVE' }));
+    }, sessionDuration);
+
+    let timeout2id = setTimeout(function() {
+      console.log(`Closing the socket forcefully 3s after graceful LEAVE`);
+      ws.close();
+    }, sessionDuration + 3000);
+
+    ws.addEventListener("close", () => {
+      clearTimeout(timeout1id);
+      clearTimeout(timeout2id);
+      console.log(`VU ${__VU}:${id}: disconnected`);
+    })
+  });
+}
+```
+
+It is important to note that no k6 iterations do not finish if there is any WebSocket still open or if a timeout/interval is not cleared or triggered. This means that you need to be careful when writing the script to clear all interval and close the websocket in some point in time. K6 will still kill the whole process if it takes too long to stop after the maximum test duration is reached.
+
+Current issues and future improvements for the WebSockets API can be found in it's [issue tracker](https://github.com/grafana/xk6-websockets/issues). Currently documentation is available through [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) although some incompatibility apply:
+- no Blob binary type - ArrayBuffer is the default
+- no `onMessage` and co. - only addEventListener
+
+
+## First class support for JavaScript Classes
+
+As part of [updating goja](https://github.com/grafana/k6/pull/2610) k6 got native, as in not by transpiling by the internal babel, support for classes 🎉. Because this actually implements classes as they described in the latest ECMAScript specification this also means we get a ton of additional class features that were never previously supported such as private fields. Additionally at least one bug [#1763](https://github.com/grafana/k6/issues/1763) was fixed as a result of this, but probably many more as well.
+
+Due to this fairly big change there are chances that some code will now behave differently. Please report any issues, though take into consideration that it's possible that the new behavior is just the correct one.
+
+Other updates from goja are:
+- optimizations around using strings and some access patterns.
+- support for `\u{01234}` unicode point syntax in regexp.
+- fixes a case where interrupting the VM would not have worked, especially around try/catch usage. This was particularly problematic for k6 as this could lead to the whole k6 hanging forever. [#2600](https://github.com/grafana/k6/pull/2600)
```

Comment:
```suggestion
- Fixed a case where interrupting the VM would not have worked, especially around try/catch usage. This was particularly problematic for k6 as this could lead to the whole k6 hanging forever. [#2600](https://github.com/grafana/k6/pull/2600)
```